src/languages/janet/index.ts

```diff
@@ -0,0 +1,199 @@
+import { usePrimaryTextOps } from "../../plugins/textOps";
+import { golfStringListLiteral, listOpsToTextOps } from "../../plugins/static";
+import {
+  applyDeMorgans,
+  bitnotPlugins,
+  equalityToInequality,
+  lowBitsPlugins,
+  pickAnyInt,
+  truncatingOpsPlugins,
+} from "../../plugins/arithmetic";
+import { forArgvToForEach } from "../../plugins/loops";
+import { alias, renameIdents } from "../../plugins/idents";
+import { assertInt64 } from "../../plugins/types";
+import { implicitlyConvertConcatArg } from "./plugins";
+
+const janetLanguage: Language = {
+  name: "Janet",
+  extension: "janet",
+  emitter: emitProgram,
+  phases: [
+    required(arraysToLists, putcToPrintChar),
+    simplegolf(golfLastPrint(false), golfLastPrintInt(true)),
+    search(
+      flipBinaryOps,
+      golfStringListLiteral(false),
+      listOpsToTextOps(),
+      equalityToInequality,
+      ...bitnotPlugins,
+      ...lowBitsPlugins,
+      applyDeMorgans,
+    ),
+
+    required(
+      usePrimaryTextOps("byte"),
+      pickAnyInt,
+      forArgvToForEach,
+      ...truncatingOpsPlugins,
+      mapOps({
+        right_align: (x) =>
+          func(
+            "string/format",
+            op("concat[Text]", text("%"), op("int_to_dec", x[1]), text("s")),
+            x[0],
+          ),
+        int_to_hex_aligned: (x) =>
+          func(
+            "string/format",
+            op("concat[Text]", text("%0"), op("int_to_dec", x[1]), text("X")),
+            x[0],
+          ),
+      }),
+    ),
+    simplegolf(implicitlyConvertConcatArg),
+    required(
+      mapOps({
+        argv: func("slice", func("dyn", builtin(":args")), int(1n)),
+
+        append: (x) => op("concat[List]", x[0], list([x[1]])),
+
+        "at[argv]": (x) =>
+          op("at[List]", func("dyn", builtin(":args")), add1(x[0])),
+        "at[byte]": (x) => op("slice[byte]", x[0], x[1], int(1n)),
+        "contains[Text]": (x) => func("int?", op("find[byte]", x[0], x[1])),
+        "contains[Table]": (x) =>
+          op("not", func("nil?", op("at[Table]", x[0], x[1]))),
+      }),
+      mapOps({
+        true: builtin("true"),
+        false: builtin("false"),
+
+        bool_to_int: (x) => conditional(x[0], int(1n), int(0n)),
+        int_to_bool: (x) => op("neq[Int]", x[0], int(0n)),
+        int_to_hex: (x) => func("string/format", text("%X"), x[0]),
+        split: (x) => func("string/split", x[1], x[0]),
+
+        "char[byte]": (x) => func("string/format", text("%c"), x[0]),
+        "concat[List]": (x) => func("array/concat", list([]), ...x),
+        "find[byte]": (x) => func("string/find", x[1], x[0]),
+        "ord[byte]": (x) => op("ord_at[byte]", x[0], int(0n)),
+        "slice[byte]": (x) =>
+          rangeIndexCall(x[0], x[1], op("add", x[1], x[2]), int(1n)),
+        "slice[List]": (x) =>
+          rangeIndexCall(x[0], x[1], op("add", x[1], x[2]), int(1n)),
+      }),
+      addMutatingInfix({ add: "+", sub: "-", mul: "*", rem: "%" }),
+      mapTo(func)({
```

Comment:
`mapToPrefixAndInfix` used in other langs has special casing for unary and binary `-` so that `((-1) * $a)` is emitted as `-a` and `((-1 * $a) + $b)` is emitted as `b-a` which won't happen using the `mapTo(func)({...})` mapping. (also, the op name mapping includes `neg` but is missing `sub`.)

Reply:
In Janet, `(- a b c d)` evaluates to `a-b-c-d` which we should be able to use.

Reply:
I got Janet working with `mapToPrefixAndInfix` (renamed to `mapUnaryAndBinary`), I'm having trouble combining that with getting Janet to do variadic emits like `(- a b c d)` but I'll add that to the golf goals for the lang